runtime/sema/check_path_expression.go

```diff
@@ -19,29 +19,44 @@
 package sema
 
 import (
+	"regexp"
+
 	"github.com/onflow/cadence/runtime/ast"
 	"github.com/onflow/cadence/runtime/common"
 )
 
 func (checker *Checker) VisitPathExpression(expression *ast.PathExpression) ast.Repr {
 
-	ty, err := CheckPathLiteral(expression)
+	ty, err := CheckPathLiteral(
+		expression.Domain.Identifier,
+		expression.Identifier.Identifier,
+		ast.NewRangeFromPositioned(expression.Domain),
+		ast.NewRangeFromPositioned(expression.Identifier),
+	)
+
 	checker.report(err)
 
 	return ty
 }
 
-func CheckPathLiteral(expression *ast.PathExpression) (Type, error) {
-
-	// Check that the domain is valid
+var isValidIdentifier = regexp.MustCompile(`^[A-Za-z_][A-Za-z0-9_]*$`).MatchString
 
-	domainIdentifier := expression.Domain
+func CheckPathLiteral(domainString, identifier string, domainRange, idRange ast.Range) (Type, error) {
 
-	domain, ok := common.AllPathDomainsByIdentifier[domainIdentifier.Identifier]
+	// Check that the domain is valid
+	domain, ok := common.AllPathDomainsByIdentifier[domainString]
 	if !ok {
 		return PathType, &InvalidPathDomainError{
-			ActualDomain: domainIdentifier.Identifier,
-			Range:        ast.NewRangeFromPositioned(domainIdentifier),
+			ActualDomain: domainString,
+			Range:        ast.NewRangeFromPositioned(domainRange),
+		}
+	}
+
+	// Check that the identifier is valid
+	if !isValidIdentifier(identifier) {
+		return PathType, &InvalidPathIdentifierError{
+			ActualIdentifier: identifier,
+			Range:            ast.NewRangeFromPositioned(idRange),
```

Comment:
The identifier range is already a range, no need to construct a new range from the existing range:

```suggestion
			Range:            idRange,
```